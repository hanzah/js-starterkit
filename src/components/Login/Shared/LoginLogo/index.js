import LoginLogo from './LoginLogo'

export default LoginLogo
