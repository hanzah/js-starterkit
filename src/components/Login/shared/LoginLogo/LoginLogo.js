import React, {PropTypes} from 'react';
import CurrentContext from 'utils/currentContext'
import ImageSelector from 'utils/imageSelector'
import styles from './LoginLogo.scss';

const Logo = ImageSelector(CurrentContext.theme, 'svgs/logo.svg');

const {string} = PropTypes;

const LoginLogo = (props) => {
  return (
    <div>
      <div className={styles.logo_containter}>
        <div className={styles.logo} data-spec='logo-wrapper'>
          <Logo data-spec='logo'/>
        </div>
        <div className={styles.product_title} data-spec='product-name'>
          {props.productName}
      </div>
      </div>
      <div className={styles.copyright_container}>
        Copyright &copy; 2017 AT&T. All rights reserved.
      </div>
    </div>
  );
}

LoginLogo.defaultProps = {
  productName: 'PRODUCT TITLE'
}

LoginLogo.propTypes = {
  productName: string
}

export default LoginLogo;