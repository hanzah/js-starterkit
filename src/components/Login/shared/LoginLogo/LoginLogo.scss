.logo_containter {
  position: absolute;
  top: 165px;
  font-family: $omnes-bold;
  font-size: 16px;
  color: rgb(255, 255, 255);
  width: 100%;
  text-align: center;
  .logo {
    >svg {
      width: 100px;
      path {
        fill: #ffffff;
      }
    }
  }
  .product_title {
    padding-top: 20px;
    letter-spacing: 2px;
  }
}

.copyright_container {
  position: absolute;
  bottom: 0;
  font-family: $omnes-regular;
  font-size: 11px;
  color: rgb(255, 255, 255);
  padding-bottom: 15px;
  text-align: center;
  width: 100%;
}