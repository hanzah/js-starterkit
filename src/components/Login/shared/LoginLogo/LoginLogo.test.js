import React from 'react';
import { shallow } from 'enzyme';

import LoginLogo from './LoginLogo';

describe('<LoginLogo/>', () => {

  it('should render a component', () => {
    const wrapper = shallow(<LoginLogo />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have Logo component', () => {
    const wrapper = shallow(<LoginLogo />);
    const Logo = wrapper.find('Logo');
    expect(Logo).toBeDefined();
  });

  it('has prop', () => {
    const wrapper = shallow(<LoginLogo />);
		expect(wrapper.props().productName).toBeUndefined();
	});
});
