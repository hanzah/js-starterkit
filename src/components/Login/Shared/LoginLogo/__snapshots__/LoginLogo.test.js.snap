exports[`<LoginLogo/> should render a component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div
        className={undefined}>
        <div
            className={undefined}
            data-spec="logo-wrapper">
            <test-file-stub
                data-spec="logo" />
        </div>
        <div
            className={undefined}
            data-spec="product-name">
            PRODUCT TITLE
        </div>
    </div>
    <div
        className={undefined}>
        Copyright © 2017 AT&T. All rights reserved.
    </div>
</div>,
  "nodes": Array [
    <div>
      <div
            className={undefined}>
            <div
                  className={undefined}
                  data-spec="logo-wrapper">
                  <test-file-stub
                        data-spec="logo" />
            </div>
            <div
                  className={undefined}
                  data-spec="product-name">
                  PRODUCT TITLE
            </div>
      </div>
      <div
            className={undefined}>
            Copyright © 2017 AT&T. All rights reserved.
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance":  {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <LoginLogo
        productName="PRODUCT TITLE" />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "productName": "PRODUCT TITLE",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <div
                    className={undefined}>
                    <div
                              className={undefined}
                              data-spec="logo-wrapper">
                              <test-file-stub
                                        data-spec="logo" />
                    </div>
                    <div
                              className={undefined}
                              data-spec="product-name">
                              PRODUCT TITLE
                    </div>
          </div>
          <div
                    className={undefined}>
                    Copyright © 2017 AT&T. All rights reserved.
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <div
                    className={undefined}>
                    <div
                              className={undefined}
                              data-spec="logo-wrapper">
                              <test-file-stub
                                        data-spec="logo" />
                    </div>
                    <div
                              className={undefined}
                              data-spec="product-name">
                              PRODUCT TITLE
                    </div>
          </div>
          <div
                    className={undefined}>
                    Copyright © 2017 AT&T. All rights reserved.
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LoginLogo
    productName="PRODUCT TITLE" />,
}
`;
